lib/doi_search.rb

```diff
@@ -35,7 +35,9 @@ def self.web_of_science(doi)
     doi = doi_name(doi)
     return [] if doi.blank?
 
-    WebOfScience.queries.user_query("DO=#{doi}").next_batch.map(&:pub_hash)
+    # filter out records that do not match the Settings.WOS.ACCEPTED_DBS filter, since only those records
+    # can be processed and added to author publication records, see https://github.com/sul-dlss/sul_pub/issues/1830
+    WebOfScience.queries.user_query("DO=#{doi}").next_batch.select { |rec| Settings.WOS.ACCEPTED_DBS.include? rec.database }.map(&:pub_hash)
```

Comment:
add the same logic as we use for processing records to when we return search results: https://github.com/sul-dlss/sul_pub/blob/main/lib/web_of_science/process_records.rb#L27